src/NHibernate/Impl/StatelessSessionImpl.cs

```diff
@@ -539,40 +539,32 @@ public void Delete(string entityName, object entity)
 			}
 		}
 
-		/// <summary> Retrieve a entity. </summary>
+		/// <summary> Retrieve an entity. </summary>
 		/// <returns> a detached entity instance </returns>
 		public object Get(string entityName, object id)
 		{
 			return Get(entityName, id, LockMode.None);
 		}
 
-		/// <summary> Retrieve a entity.
-		///
+		/// <summary>
+		/// Retrieve an entity.
 		/// </summary>
 		/// <returns> a detached entity instance
 		/// </returns>
 		public T Get<T>(object id)
 		{
-			using (BeginProcess())
-			{
-				return (T)Get(typeof(T), id);
-			}
-		}
-
-		private object Get(System.Type persistentClass, object id)
-		{
-			return Get(persistentClass.FullName, id);
+			return (T) Get(typeof(T).FullName, id);
 		}
 
 		/// <summary>
-		/// Retrieve a entity, obtaining the specified lock mode.
+		/// Retrieve an entity, obtaining the specified lock mode.
 		/// </summary>
 		/// <returns> a detached entity instance </returns>
 		public object Get(string entityName, object id, LockMode lockMode)
 		{
 			using (BeginProcess())
 			{
-				object result = Factory.GetEntityPersister(entityName).Load(id, null, lockMode, this);
+				object result = Factory.GetEntityPersister(entityName).Load(id, null, lockMode ?? LockMode.None, this);
```

Comment:
For consistency with `session.Get(entityName, id, lockMode)`

Reply:
Did you mean `Get(string entityName, object id)`? Otherwise I do not get it. Well I do not get it either, I see no consistency issue here.

Reply:
For consistency with call `session.Get(entityName, id, null)` which uses `LockMode.None` by default. Stateless session for my surprise was throwing exception

Reply:
Ok.